src/Microsoft.Android.Runtime.NativeAOT/Android.Runtime.NativeAOT/TypeMapping.cs

```diff
@@ -0,0 +1,77 @@
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Hashing;
+using System.Runtime.InteropServices;
+using System.Text;
+using Android.Runtime;
+
+namespace Microsoft.Android.Runtime;
+
+internal static class TypeMapping
+{
+	private static Dictionary<Type, string[]> s_javaClassNames = new ();
+
+	internal static bool TryGetType (string javaClassName, [NotNullWhen (true)] out Type? type)
+	{
+		ulong hash = Hash (javaClassName);
+
+		// the hashes array is sorted and all the hashes are unique
+		int index = MemoryExtensions.BinarySearch (Hashes, hash);
+		if (index < 0) {
+			type = null;
+			return false;
+		}
+
+		// ensure this is not a hash collision
+		string resolvedJavaClassName = GetJavaClassNameByIndex (index);
+		if (!resolvedJavaClassName.Equals (javaClassName, StringComparison.Ordinal)) {
```

Comment:
```suggestion
		if (resolvedJavaClassName != javaClassName) {
```
Faster and more concise way to do the same.